documenting/styleguide.rst

```diff
@@ -0,0 +1,220 @@
+
+.. _style-guide:
+
+Style Guide
+===========
+
+.. highlight::  rest
+
+Use of whitespace
+-----------------
+
+All reST files use an indentation of 3 spaces; no tabs are allowed.  The
+maximum line length is 80 characters for normal text, but tables, deeply
+indented code samples and long links may extend beyond that.  Code example
+bodies should use normal Python 4-space indentation.
+
+Make generous use of blank lines where applicable; they help group things
+together.
+
+A sentence-ending period may be followed by one or two spaces; while reST
+ignores the second space, it is customarily put in by some users, for example
+to aid Emacs' auto-fill mode.
+
+Footnotes
+---------
+
+Footnotes are generally discouraged, though they may be used when they are the
+best way to present specific information. When a footnote reference is added at
+the end of the sentence, it should follow the sentence-ending punctuation. The
+reST markup should appear something like this::
+
+    This sentence has a footnote reference. [#]_ This is the next sentence.
+
+Footnotes should be gathered at the end of a file, or if the file is very long,
+at the end of a section. The docutils will automatically create backlinks to
+the footnote reference.
+
+Footnotes may appear in the middle of sentences where appropriate.
+
+Capitalization
+--------------
+
+.. sidebar:: Sentence case
+
+   Sentence case is a set of capitalization rules used in English
+   sentences: the first word is always capitalized and other words are
+   only capitalized if there is a specific rule requiring it.
+
+In the Python documentation, the use of sentence case in section titles is
+preferable, but consistency within a unit is more important than
+following this rule.  If you add a section to a chapter where most
+sections are in title case, you can either convert all titles to
+sentence case or use the dominant style in the new section title.
```

Comment:
Re: this advice:

At least for the files you're moving around, shall we convert the titles to sentence case?

Reply:
If we had the choice now, sentence case would be preferred for consistency with the Python docs, but I'm not sure its worth the churn changing at this point. We should be consistent, though (as this section says), and there are some that use sentence case, and at least one title that mixes both.